docs/source/general/contributing.rst

```diff
@@ -16,6 +16,44 @@ If you intend to contribute in any way please keep the following guidelines in m
 8. Document your changes
 9. Send pull request
 
+
+Virtual Environment
+-------------------
+
+Please use a virtual environment while developing your feature, it keeps everybody on the same page and it helps reproducing bugs and resolving problems.
+
+.. code-block:: shell
+
+    virtualenv env
+    source env/bin/activate
+    python setup.py install
+
+Pre Commit Niceties
+-------------------
+
+Install ``flake8`` to check for common pitfalls that may have your contribution stopped
+
+.. code-block:: shell
+
+    source env/bin/activate
+    pip install flake8
+
+Please use ``runflake8`` before committing your work and opening a pull request
+
+.. note::
+    To speed things up you can add your virtual environment directory to the list of the excluded directories in the ``runflake8`` script
```

Comment:
it would be even better to use [virtualenvwrapper](https://virtualenvwrapper.readthedocs.io/en/latest/) which wouldn't give you this additional problem. Check it out.

Virtualenv management and creation would then become:

```
# create virtualenv
mkvirtualenv --python=python3 netjsonconfig
# activate virtualenv
workon netjsonconfig
# deactivate virtualenv
deactivate
# remove virtualenv
rmvirtualenv netjsonconfig
```

I leave it up to you to change the text to suggest mkvirtualenv or leave it as it is now.